engine/src/main/java/org/terasology/logic/health/DoDestroyEvent.java

```diff
@@ -36,11 +38,36 @@ public EntityRef getInstigator() {
         return instigator;
     }
 
+    public String getInstigatorString() {
+        if (instigator.getParentPrefab() != null) {
+            String instigatorString = instigator.getParentPrefab().getName();
+            //getParentPrefab.getName() returns a ResourceUrn String such as "engine:player"
+            //The following calls change the damage type to be more readable
+            //For instance, "engine:player" becomes "Player"
+            instigatorString = instigatorString.replaceAll(".*:(.*)", "$1");
+            instigatorString = Character.toUpperCase(instigatorString.charAt(0)) + instigatorString.substring(1);
+            return instigatorString;
+        } else {
+            return null;
+        }
+    }
+
     public EntityRef getDirectCause() {
         return directCause;
     }
 
     public Prefab getDamageType() {
         return damageType;
     }
+
+    public String getDamageTypeString() {
```

Comment:
Same as above: We should support a display name component for damage types before falling back to the prefab name.
Assuming a module registers a damage type like `InternalBossEventPoisonDamage`, this is nothing we want to show the player directly.
Instead, the prefab for such a damage could look like this:
```
{
	"DisplayName" : {
		"name" : "Deadly Poison Aura"
	}
}

```
We can still keep the fallback code for prefabs without a name and maybe log a info that the damage type is missing a readable name. As player deaths should not occur multiple times per second, this would not create too much noise in the logs.
This would also leave an option to i18n the damage type names.